user_guide/src/examples/dataframe/dataset.py

```diff
@@ -0,0 +1,15 @@
+from .guide_magic import my_datetime_list
+import polars as pl
+
+
+N = 1000
+
+df = pl.DataFrame(
+    {
+        "rank": list(range(N)),
+        "id": list(map(lambda x: str(x).zfill(5), range(N, 0, -1))),
```

Comment:
I would prefer a `pl.arange` and a following `with_column` here to use polars expressions to prepare the data, not python lambdas. 

One of our selling points, is that you need them less.

Reply:
How's this?

```py
df = pl.DataFrame({"date": my_datetime_list}).with_columns(
    [
        pl.arange(0, N).alias("rank"),
        pl.arange(0, N).reverse().alias("id"),
    ]
)
```
What would be the best method for implementing the `zfill` here? So

59 -> `zfill(5)` -> 00059

Reply:
Is it just that we want to use `pl.Expr.apply`?

```py
pl.col("id").apply(lambda x: str(x).zfill(5))
```

Originally I was just building a DataFrame with library-agnostic methods as a soft introduction, but incorporating more surface area sounds better to me.

Reply:
You're fast https://github.com/pola-rs/polars/pull/3593

So I guess something like this in the next release?

```py
df = pl.DataFrame({"date": my_datetime_list})
df = df.with_columns(
    [
        pl.arange(0, N).alias("rank"),
        pl.arange(0, N).reverse().cast(pl.Utf8).str.zfill(5).alias("id"),
    ]
)
```

Reply:
Yes, And maybe we can do this even? 

```python
 pl.arange(pl.count(), 0, -1).cast(pl.Utf8).str.zfill(5).alias("id"),
```

Note that with `pl.count()` you don't need to know the length of you dataframe, and thus this works also in lazy chaining.

Reply:
> ```python
>  pl.arange(pl.count(), 0, -1).cast(pl.Utf8).str.zfill(5).alias("id"),
> ```

this gives me 
```
OverflowError: can't convert negative int to unsigned
```